File: src/main/java/javabot/operations/QuitOperation.java
package javabot.operations;

import java.util.ArrayList;
import java.util.List;

import javabot.BotEvent;
import javabot.Message;
import javabot.Javabot;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class QuitOperation extends BotOperation {
    private static final Logger log = LoggerFactory.getLogger(QuitOperation.class);

    public QuitOperation(final Javabot javabot) {
        super(javabot);
    }

    @Override
    public boolean handleMessage(final BotEvent event) {
        final List<Message> messages = new ArrayList<Message>();
        final String message = event.getMessage();
        if(message.toLowerCase().startsWith("quit ")) {
            if(message.substring("quit ".length()).equals(getBot().getNickPassword())) {
                if(log.isDebugEnabled()) {
                    log.debug("About to quit");
                }
                System.exit(0);
            }
        }
        return false;
    }
}

File: src/main/kotlin/javabot/operations/QuitOperation.kt
package javabot.operations

import jakarta.inject.Inject
import java.util.Locale
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.ConfigDao

class QuitOperation
@Inject
constructor(bot: Javabot, adminDao: AdminDao, var configDao: ConfigDao) :
    BotOperation(bot, adminDao) {

    override fun handleMessage(event: Message): List<Message> {
        val message = event.value
        if (message.lowercase(Locale.getDefault()).startsWith("quit ")) {
            if (message.substring("quit ".length) == configDao.get().password) {
                System.exit(0)
            }
        }
        return listOf()
    }
}
